test/storage/sql_table_test.cpp

```diff
@@ -246,105 +316,677 @@ class SqlTableRW {
+  value = row1.AccessWithNullCheck(pc_pair.second.at(catalog::col_oid_t(2)));
+  uint32_t new_col = *reinterpret_cast<uint32_t *>(value);
+  if (id != 400) {
+    EXPECT_EQ(new_col, new_col_default_value);
+  } else {
+    EXPECT_EQ(new_col, new_col_map[id]);
+  }
+
+  // check the if we get (200, 10001, null)
+  auto row2 = pc->InterpretAsRow(1);
+  value = row2.AccessWithNullCheck(pc_pair.second.at(catalog::col_oid_t(0)));
+  EXPECT_NE(value, nullptr);
+  id = *reinterpret_cast<uint32_t *>(value);
+  EXPECT_FALSE(seen_map[id]);
+  seen_map[id] = true;
+  value = row2.AccessWithNullCheck(pc_pair.second.at(catalog::col_oid_t(1)));
+  EXPECT_NE(value, nullptr);
+  datname = *reinterpret_cast<uint32_t *>(value);
+  EXPECT_EQ(datname, datname_map[id]);
+  value = row2.AccessWithNullCheck(pc_pair.second.at(catalog::col_oid_t(2)));
+  new_col = *reinterpret_cast<uint32_t *>(value);
+  if (id != 400) {
+    EXPECT_EQ(new_col, new_col_default_value);
+  } else {
+    EXPECT_EQ(new_col, new_col_map[id]);
+  }
+
+  pc = pc_pair.first.Initialize(buffer);
+  // Need to scan again to get the rest of the data
+  table.table_->Scan(txn, &start_pos, pc, pc_pair.second, table.version_);
+  EXPECT_EQ(pc->NumTuples(), 2);
+  EXPECT_TRUE(start_pos == table.table_->end());
+
+  auto row3 = pc->InterpretAsRow(0);
+  value = row3.AccessWithNullCheck(pc_pair.second.at(catalog::col_oid_t(0)));
+  EXPECT_NE(value, nullptr);
+  id = *reinterpret_cast<uint32_t *>(value);
+  EXPECT_FALSE(seen_map[id]);
+  seen_map[id] = true;
+  value = row3.AccessWithNullCheck(pc_pair.second.at(catalog::col_oid_t(1)));
+  EXPECT_NE(value, nullptr);
+  datname = *reinterpret_cast<uint32_t *>(value);
+  EXPECT_EQ(datname, datname_map[id]);
+  value = row3.AccessWithNullCheck(pc_pair.second.at(catalog::col_oid_t(2)));
+  new_col = *reinterpret_cast<uint32_t *>(value);
+  if (id != 400) {
+    EXPECT_EQ(new_col, new_col_default_value);
+  } else {
+    EXPECT_EQ(new_col, new_col_map[id]);
+  }
+
+  // check the if we get (200, 10001, null)
+  auto row4 = pc->InterpretAsRow(1);
+  value = row4.AccessWithNullCheck(pc_pair.second.at(catalog::col_oid_t(0)));
+  EXPECT_NE(value, nullptr);
+  id = *reinterpret_cast<uint32_t *>(value);
+  EXPECT_FALSE(seen_map[id]);
+  seen_map[id] = true;
+  value = row4.AccessWithNullCheck(pc_pair.second.at(catalog::col_oid_t(1)));
+  EXPECT_NE(value, nullptr);
+  datname = *reinterpret_cast<uint32_t *>(value);
+  EXPECT_EQ(datname, datname_map[id]);
+  value = row4.AccessWithNullCheck(pc_pair.second.at(catalog::col_oid_t(2)));
+  new_col = *reinterpret_cast<uint32_t *>(value);
+  if (id != 400) {
+    EXPECT_EQ(new_col, new_col_default_value);
+  } else {
+    EXPECT_EQ(new_col, new_col_map[id]);
+  }
+
+  for (auto iter : seen_map) EXPECT_TRUE(iter.second);
+
+  delete[] buffer;
+  txn_manager_.Commit(txn, TestCallbacks::EmptyCallback, nullptr);
+  delete txn;
 }
 
 // NOLINTNEXTLINE
 TEST_F(SqlTableTests, VarlenInsertTest) {
```

Comment:
I think you might want to open GC in tests to see whether your implementation works well with GC, to ensure you are
(1) not double-reclaiming memory (2) retaining useful contents(specially varlen contents) from reclaiming by GC.
One varlen insert seems too simple here. You can test updating an old-of-date row with varlen contents.
This comment comes from the doubt I mentioned in https://github.com/cmu-db/terrier/pull/342/files?file-filters%5B%5D=.cpp&file-filters%5B%5D=.h&file-filters%5B%5D=.py#r283081606